tools/travis-before-install.sh

```diff
@@ -33,5 +36,11 @@ python -V
 popd
 
 pip install --upgrade pip setuptools
+# pycparser 2.19 doesn't install correctly when Python's optimization level is
+# set to strip docstrings (see https://github.com/eliben/pycparser/issues/291),
+# so preemptively install pycparser with PYTHONOPTIMIZE unset.  If a future
+# version of pycparser doesn't have this problem, the following line can be
+# removed.
+PYTHONOPTIMIZE="" pip install -v pycparser
 pip install -r test_requirements.txt
```

Comment:
Why not combine this with the next line `PYTHONOPTIMIZE= pip install -r test_requirements.txt`

Reply:
That turns off the optimize flags for all the packages in `test_requirements.txt`.  I guess that would OK.  Presumably the intent of using `PYTHONOPTIMIZE=2` is to test *numpy* when built with that setting, so we don't care how the packages in `test_requirements.txt` are built.